pkg/crd/spec.go

```diff
@@ -151,14 +160,18 @@ func (p *Parser) NeedCRDFor(groupKind schema.GroupKind, maxDescLen *int) {
 	hasStorage := false
 	for _, ver := range crd.Spec.Versions {
 		if ver.Storage {
+			// more than one storage, already set it true on previous iteration
+			if hasStorage {
+				errs = append(errs, fmt.Errorf("CRD for %s has more than one storage version", groupKind))
+				break
+			}
 			hasStorage = true
-			break
 		}
 	}
 	if !hasStorage {
 		// just add the error to the first relevant package for this CRD,
 		// since there's no specific error location
-		packages[0].AddError(fmt.Errorf("CRD for %s has no storage version", groupKind))
+		errs = append(errs, fmt.Errorf("CRD for %s has no storage version", groupKind))
```

Comment:
why was this changed?

Reply:
@vincepri  I was following this PR 
https://github.com/kubernetes-sigs/controller-tools/pull/399/files#diff-05d11052b3234e4c8ce76c9362508cbd5c3f275a76ee2513240234ed879cb2b5L151-R163 

Should I stick with that approach or the one you suggested?

Reply:
@vincepri  I wouldn't recommend to pass the package into the function because  it would complicate the unit test writing ( need to setup a package loader ) . I would keep the function loosely coupled with that